platforms/documentation/docs/src/docs/userguide/authoring-builds/plugins/testing_gradle_plugins.adoc

```diff
@@ -313,6 +302,53 @@ TestKit determines the plugin classpath by running a specific Gradle task.
 You will need to execute the `assemble` task to initially generate the plugin classpath or to reflect changes to it even when running TestKit-based functional tests from the IDE.
 
 Some IDEs provide a convenience option to delegate the "test classpath generation and execution" to the build.
-In IntelliJ you can find this option under _Preferences... > Build, Execution, Deployment > Build Tools > Gradle > Runner > Delegate IDE build/run actions to gradle_.
+In IntelliJ, you can find this option under Preferences... > Build, Execution, Deployment > Build Tools > Gradle > Runner > Delegate IDE build/run actions to Gradle.
 
 image::intellij-delegate-to-build.png[]
+
+[[sec:writing_tests_for_your_plugin]]
+== Writing tests for your plugin
+
+You can use the link:{javadocPath}/org/gradle/testfixtures/ProjectBuilder.html[ProjectBuilder] class to create link:{groovyDslPath}/org.gradle.api.Project.html[Project] instances to use when you test your plugin implementation.
+
+=== Example: Testing a custom plugin
+
+[source.multi-language-sample,java]
+.src/test/java/org/example/GreetingPluginTest.java
+----
+include::{snippetsPath}/plugins/customPlugin/groovy/java-gradle-plugin/src/test/java/org/example/GreetingPluginTest.java[tag=test-plugin]
+----
+
+Plugins often also provide custom task types. Please see <<custom_tasks.adoc#custom_tasks,Developing Custom Gradle Task Types>> for more details.
+
+Gradle provides a number of features that are helpful when developing Gradle types, including plugins. Please see <<custom_gradle_types.adoc#custom_gradle_types,Developing Custom Gradle Types>> for more details.
```

Comment:
❗ Everything under here seems incomplete, or like it was meant to be moved somewhere else?

Reply:
This is all the old docs. It's always been that way. I'm going to take a pass at updating the Testing section.

Reply:
Looks like a copy paste issue.